src/Native/System.Native/pal_networking.cpp

```diff
@@ -2329,8 +2343,13 @@ static Error CloseSocketEventPortInner(int32_t port)
 static Error TryChangeSocketEventRegistrationInner(
     int32_t port, int32_t socket, SocketEvents currentEvents, SocketEvents newEvents, uintptr_t data)
 {
+#ifdef EV_RECEIPT
     const uint16_t AddFlags = EV_ADD | EV_CLEAR | EV_RECEIPT;
     const uint16_t RemoveFlags = EV_DELETE | EV_RECEIPT;
+#else
+    const uint16_t AddFlags = EV_ADD | EV_CLEAR;
+    const uint16_t RemoveFlags = EV_DELETE;
```

Comment:
In some corner cases, it will not return the error.

Currently, go_lang for example set it to zero for NetBSD defs: https://golang.org/src/runtime/defs_netbsd.go.

This flag was not present in NetBSD and DragonFly until very recently when it was added few weeks ago: https://github.com/facebook/fb-adb/issues/38 -> https://github.com/IIJ-NetBSD/netbsd-src/commit/fe672be45b288bad28419f6b8f96214435f2ad5f.

We can also define it to zero, but it will have equal effect than ifdef'ing it out for time being IMO (until the next OS release).